packages/components/src/Banner/Banner.css

```diff
@@ -1,27 +1,63 @@
 :root {
   --flash--inset: var(--space-base);
+  --flash--inset--small: var(--space-small);
 }
 
 .flash {
   display: flex;
-  margin-bottom: var(--space-base);
-  padding: var(--flash--inset);
-  border-top: var(--border-thick) solid var(--color-grey--light);
-  border-bottom-right-radius: var(--space-smaller);
-  border-bottom-left-radius: var(--space-smaller);
+  box-shadow: var(--shadow-base);
+  padding: var(--flash--inset--small) var(--flash--inset);
   color: var(--color-black);
   background-color: var(--color-grey--lightest);
-  align-items: center;
+  align-items: start;
+}
+
+.flash.medium {
+  padding: var(--flash--inset);
 }
 
-.flash > *:first-child {
+.bannerContent {
+  display: flex;
   flex: 1 1 auto;
+  flex-wrap: wrap;
+  justify-content: space-between;
+}
+
+.bannerContent > * {
+  margin-top: var(--flash--inset--small);
+  margin-bottom: var(--flash--inset--small);
+}
+
+.bannerContent > *:first-child {
+  margin-right: var(--space-base);
+}
+
+.medium .bannerContent {
+  flex-wrap: nowrap;
+}
+
+.medium .bannerContent > * {
+  margin-top: 0;
+  margin-bottom: 0;
+}
+
+.bannerAction {
+  display: flex;
+  flex: 0 0 auto;
+}
+
+.dismissWrapper {
+  display: flex;
+  position: relative;
+  height: var(--space-large);
+  padding-left: var(--flash--inset);
+  align-self: stretch;
 }
 
 .closeButton {
-  width: var(--space-larger);
-  margin-top: calc(var(--space-smallest) - var(--flash--inset));
-  margin-right: calc(var(--space-smallest) - var(--flash--inset));
+  position: absolute;
```

Comment:
I feel this should be on the `dismissWrapper` instead of the button. That way we aren't leaving an empty `div` in the DOM flow